### DISCUSSION TITLE
How to Add Line Break into Documentation Comments

### DISCUSSION THREAD

**Discussion by wachidsusilo** [microsoft/vscode-cpptools#10243] [General]
How to add a new line in C++ documentation comments?
I have a long documentation comments like this:
```cpp

/**
 * @brief Parse the given String into an Any object. 
 * If the String starts with '{' and ends with '}' it will be parsed as an Object.
 * If the String starts with '[' and ends with ']' it will be parsed as an Array.
 * If the String starts with '"' and ends with '"' it will be parsed as a String.
 * If the String is a number it will be parsed as a Number.
 * If the String is 'true' or 'false' it will be parsed as a Boolean.
 * Otherwise it will be parsed as a Null.
 * 
 * NOTE: If the String is an Object, the returned Any object will be a String.
 * The String will be deserialized when the object is accessed. The reasoning behind 
 * this is that at the time of parsing, the type of the Object is not known.
 *
 * @param str the String to parse.
 * @return The parsed Any object.
 */
Any Any::parse(const String &str) {}
```

But the new line is removed and make the documentation hard to read:

> Parse the given String into an Any object. If the String starts with '{' and ends with '}' it will be parsed as an Object. If the String starts with '[' and ends with ']' it will be parsed as an Array. If the String starts with '"' and ends with '"' it will be parsed as a String. If the String is a number it will be parsed as a Number. If the String is 'true' or 'false' it will be parsed as a Boolean. Otherwise it will be parsed as a Null. NOTE: If the String is an Object, the returned Any object will be a String. The String will be deserialized when the object is accessed. The reasoning behind this is that at the time of parsing, the type of the Object is not known.
> 
> **Parameters:**
> `str` – the String to parse.
> 
> **Returns:**
> The parsed Any object.

**Comment by sean-mcmanus** [comment]
We're tracking this with https://github.com/microsoft/vscode-cpptools/issues/10039 . You can upvote it.

Or set C_Cpp.simplifyStructuredComments to false.

**Reply by wachidsusilo** [reply]
@sean-mcmanus Thanks for the quick reply. 

> We're tracking this with https://github.com/microsoft/vscode-cpptools/issues/10039 . You can upvote it.

What do you mean by 'upvote it'? How to 'upvote' an issue?

> Or set C_Cpp.simplifyStructuredComments to false.

This doesn't work though. It still removes the new lines.

> @brief  – Parse the given String into an Any object. If the String starts with '{' and ends with '}' it will be parsed as an Object. If the String starts with '[' and ends with ']' it will be parsed as an Array. If the String starts with '"' and ends with '"' it will be parsed as a String. If the String is a number it will be parsed as a Number. If the String is 'true' or 'false' it will be parsed as a Boolean. Otherwise it will be parsed as a Null. NOTE: If the String is an Object, the returned Any object will be a String. The String will be deserialized when the object is accessed. The reasoning behind this is that at the time of parsing, the type of the Object is not known.
> 
> *@param* `str` – is the String to parse.
> 
> *@return* – The parsed Any object.

And besides that, I want the hint to be formatted, because the docs are for the user of the library.  I've tried putting `\n` and still not working.
